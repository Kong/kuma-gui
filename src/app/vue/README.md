---
section: Packages
---
# vue

Automatic `vue` initialization using our projects dependency injection layer.

