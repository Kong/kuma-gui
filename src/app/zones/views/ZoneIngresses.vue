<template>
  <div class="zoneingresses">
    <MultizoneInfo v-if="multicluster === false" />

    <!-- Zone CPs information for when Multicluster is enabled -->
    <FrameSkeleton v-else>
      <DataOverview
        :selected-entity-name="entity?.name"
        :page-size="pageSize"
        :is-loading="isLoading"
        :error="error"
        :empty-state="empty_state"
        :table-data="tableData"
        :table-data-is-empty="isEmpty"
        :next="next"
        :page-offset="pageOffset"
        @table-action="tableAction"
        @load-data="loadData"
      >
        <template #additionalControls>
          <KButton
            v-if="$route.query.ns"
            class="back-button"
            appearance="primary"
            icon="arrowLeft"
            :to="{ name: 'zoneingresses' }"
          >
            View all
          </KButton>
        </template>
      </DataOverview>
      <TabsWidget
        v-if="isEmpty === false"
        :has-error="error !== null"
        :is-loading="isLoading"
        :tabs="tabs"
        initial-tab-override="overview"
      >
        <template #tabHeader>
          <h1 class="entity-heading">
            Zone Ingress: {{ entity.name }}
          </h1>
        </template>
        <template #overview>
          <LabelList>
            <div>
              <ul>
                <li
                  v-for="(value, key) in entity"
                  :key="key"
                >
                  <h4>
                    {{ key }}
                  </h4>
                  <p>
                    {{ value }}
                  </p>
                </li>
              </ul>
            </div>
          </LabelList>
        </template>
        <template #insights>
          <KCard border-variant="noBorder">
            <template #body>
              <AccordionList :initially-open="0">
                <AccordionItem
                  v-for="(value, key) in subscriptionsReversed"
                  :key="key"
                >
                  <template #accordion-header>
                    <SubscriptionHeader :details="value" />
                  </template>

                  <template #accordion-content>
                    <SubscriptionDetails
                      :details="value"
                      is-discovery-subscription
                    />
                  </template>
                </AccordionItem>
              </AccordionList>
            </template>
          </KCard>
        </template>

        <template #xds-configuration>
          <EnvoyData
            data-path="xds"
            :zone-ingress-name="entity.name"
            query-key="envoy-data-zone-ingress"
          />
        </template>

        <template #envoy-stats>
          <EnvoyData
            data-path="stats"
            :zone-ingress-name="entity.name"
            query-key="envoy-data-zone-ingress"
          />
        </template>

        <template #envoy-clusters>
          <EnvoyData
            data-path="clusters"
            :zone-ingress-name="entity.name"
            query-key="envoy-data-zone-ingress"
          />
        </template>
      </TabsWidget>
    </FrameSkeleton>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import { KButton, KCard } from '@kong/kongponents'

import { getItemStatusFromInsight } from '@/utilities/dataplane'
import { getSome } from '@/utilities/helpers'
import { kumaApi } from '@/api/kumaApi'
import { PAGE_SIZE_DEFAULT } from '@/constants'
import { QueryParameter } from '@/utilities/QueryParameter'
import AccordionItem from '@/app/common/AccordionItem.vue'
import AccordionList from '@/app/common/AccordionList.vue'
import DataOverview from '@/app/common/DataOverview.vue'
import EnvoyData from '@/app/common/EnvoyData.vue'
import FrameSkeleton from '@/app/common/FrameSkeleton.vue'
import LabelList from '@/app/common/LabelList.vue'
import MultizoneInfo from '../components/MultizoneInfo.vue'
import SubscriptionDetails from '@/app/common/subscriptions/SubscriptionDetails.vue'
import SubscriptionHeader from '@/app/common/subscriptions/SubscriptionHeader.vue'
import TabsWidget from '@/app/common/TabsWidget.vue'

export default {
  name: 'ZoneIngresses',

  components: {
    AccordionItem,
    AccordionList,
    DataOverview,
    EnvoyData,
    FrameSkeleton,
    LabelList,
    MultizoneInfo,
    SubscriptionDetails,
    SubscriptionHeader,
    TabsWidget,
    KButton,
    KCard,
  },

  props: {
    selectedZoneIngressName: {
      type: String,
      required: false,
      default: null,
    },

    offset: {
      type: Number,
      required: false,
      default: 0,
    },
  },

  data() {
    return {
      isLoading: true,
      isEmpty: false,
      error: null,

      empty_state: {
        title: 'No Data',
        message: 'There are no Zone Ingresses present.',
      },
      tableData: {
        headers: [
          { key: 'actions', hideLabel: true },
          { label: 'Status', key: 'status' },
          { label: 'Name', key: 'name' },
        ],
        data: [],
      },
      tabs: [
        {
          hash: '#overview',
          title: 'Overview',
        },
        {
          hash: '#insights',
          title: 'Zone Ingress Insights',
        },
        {
          hash: '#xds-configuration',
          title: 'XDS Configuration',
        },
        {
          hash: '#envoy-stats',
          title: 'Stats',
        },
        {
          hash: '#envoy-clusters',
          title: 'Clusters',
        },
      ],
      entity: {},
      rawData: [],
      pageSize: PAGE_SIZE_DEFAULT,
      next: null,
      subscriptionsReversed: [],
      pageOffset: this.offset,
    }
  },

  computed: {
    ...mapGetters({
      multicluster: 'config/getMulticlusterStatus',
    }),
  },

  watch: {
    $route() {
      // Ensures basic state is reset when switching meshes using the mesh selector.
      this.isLoading = true
      this.isEmpty = false
      this.error = null

      this.init(0)
    },
  },

  beforeMount() {
    this.init(this.offset)
  },

  methods: {
    init(offset) {
      if (this.multicluster) {
        this.loadData(offset)
      }
    },
    tableAction(ev) {
      const data = ev

      // load the data into the tabs
      this.getEntity(data)
    },

    async loadData(offset) {
      this.pageOffset = offset
      // Puts the offset parameter in the URL so it can be retrieved when the user reloads the page.
      QueryParameter.set('offset', offset > 0 ? offset : null)

      this.isLoading = true
      this.isEmpty = false

      const name = this.$route.query.ns || null
      const size = this.pageSize

      try {
        const { data, next } = await this.getZoneIngressOverviews(name, size, offset)

        // set pagination
        this.next = next

        // set table data
        if (data.length) {
          this.isEmpty = false
          this.rawData = data
          this.getEntity({ name: this.selectedZoneIngressName ?? data[0].name })

          this.tableData.data = data.map((item) => {
            const { zoneIngressInsight = {} } = item
            const status = getItemStatusFromInsight(zoneIngressInsight)

            return { ...item, status }
          })
        } else {
          this.tableData.data = []
          this.isEmpty = true
        }
      } catch (err) {
        if (err instanceof Error) {
          this.error = err
        } else {
          console.error(err)
        }

        this.isEmpty = true
      } finally {
        this.isLoading = false
      }
    },

    getEntity(entity) {
      const selected = ['type', 'name']
      const item = this.rawData.find((data) => data.name === entity.name)

      const subscriptions = item?.zoneIngressInsight?.subscriptions ?? []

      this.subscriptionsReversed = Array.from(subscriptions).reverse()

      this.entity = getSome(item, selected)
      QueryParameter.set('zoneIngress', this.entity.name)
    },

    /**
     * @param {string | null} name
     * @param {number} size
     * @param {number} offset
     * @returns {Promise<{ data: ZoneIngressOverview[], next: string | null }>}
     */
    async getZoneIngressOverviews(name, size, offset) {
      if (name) {
        const zoneIngressOverview = await kumaApi.getZoneIngressOverview({ name }, { size, offset })

        return {
          data: [zoneIngressOverview],
          next: null,
        }
      } else {
        const { items, next } = await kumaApi.getAllZoneIngressOverviews({ size, offset })

        return {
          data: items ?? [],
          next,
        }
      }
    },
  },
}
</script>
