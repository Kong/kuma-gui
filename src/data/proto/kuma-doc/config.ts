/* eslint-disable */

export const protobufPackage = "doc";

export interface Config {
  type: Config_Type;
  name: string;
  fileName: string;
  templateName: string;
}

export enum Config_Type {
  Undefined = "Undefined",
  Policy = "Policy",
  Proxy = "Proxy",
  Other = "Other",
  UNRECOGNIZED = "UNRECOGNIZED",
}

export function config_TypeFromJSON(object: any): Config_Type {
  switch (object) {
    case 0:
    case "Undefined":
      return Config_Type.Undefined;
    case 1:
    case "Policy":
      return Config_Type.Policy;
    case 2:
    case "Proxy":
      return Config_Type.Proxy;
    case 99:
    case "Other":
      return Config_Type.Other;
    case -1:
    case "UNRECOGNIZED":
    default:
      return Config_Type.UNRECOGNIZED;
  }
}

export function config_TypeToJSON(object: Config_Type): string {
  switch (object) {
    case Config_Type.Undefined:
      return "Undefined";
    case Config_Type.Policy:
      return "Policy";
    case Config_Type.Proxy:
      return "Proxy";
    case Config_Type.Other:
      return "Other";
    case Config_Type.UNRECOGNIZED:
    default:
      return "UNRECOGNIZED";
  }
}

function createBaseConfig(): Config {
  return { type: Config_Type.Undefined, name: "", fileName: "", templateName: "" };
}

export const Config = {
  fromJSON(object: any): Config {
    return {
      type: isSet(object.type) ? config_TypeFromJSON(object.type) : Config_Type.Undefined,
      name: isSet(object.name) ? String(object.name) : "",
      fileName: isSet(object.fileName) ? String(object.fileName) : "",
      templateName: isSet(object.templateName) ? String(object.templateName) : "",
    };
  },

  toJSON(message: Config): unknown {
    const obj: any = {};
    message.type !== undefined && (obj.type = config_TypeToJSON(message.type));
    message.name !== undefined && (obj.name = message.name);
    message.fileName !== undefined && (obj.fileName = message.fileName);
    message.templateName !== undefined && (obj.templateName = message.templateName);
    return obj;
  },

  fromPartial(object: DeepPartial<Config>): Config {
    const message = Object.create(createBaseConfig()) as Config;
    message.type = object.type ?? Config_Type.Undefined;
    message.name = object.name ?? "";
    message.fileName = object.fileName ?? "";
    message.templateName = object.templateName ?? "";
    return message;
  },
};

type Builtin = Date | Function | Uint8Array | string | number | boolean | undefined;

export type DeepPartial<T> = T extends Builtin ? T
  : T extends Array<infer U> ? Array<DeepPartial<U>> : T extends ReadonlyArray<infer U> ? ReadonlyArray<DeepPartial<U>>
  : T extends {} ? { [K in keyof T]?: DeepPartial<T[K]> }
  : Partial<T>;

function isSet(value: any): boolean {
  return value !== null && value !== undefined;
}
