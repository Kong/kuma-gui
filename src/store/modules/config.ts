import Kuma from '@/services/kuma'
import { StoreOptions, ActionTree, GetterTree, MutationTree } from 'vuex'
import { RootInterface } from '..'

export interface ConfigInterface {
  status: string | null
  environment: string | null
}

export type ConfigType = StoreOptions<ConfigInterface>

const state: ConfigInterface = {
  status: null,
  environment: null,
}

const mutations: MutationTree<ConfigInterface> = {
  SET_STATUS: (state, status: string) => (state.status = status),
  SET_ENVIRONMENT: (state, value) => (state.environment = value),
}

const getters: GetterTree<ConfigInterface, RootInterface> = {
  getStatus: (state) => state.status,
  getEnvironment: (state) => state.environment,
}

const actions:(api: Kuma) => ActionTree<ConfigInterface, RootInterface> = (api: Kuma) => ({
  // get the status of the API
  getStatus ({ commit }) {
    return api.getStatus()
      .then(response => {
        commit('SET_STATUS', response)
      })
  },
})

export default (api: Kuma) => ({
  namespaced: true,
  state,
  getters,
  mutations,
  actions: actions(api)
})
