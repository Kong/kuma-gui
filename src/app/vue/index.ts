import { createApp } from 'vue'
import {
  createRouter,
  createWebHistory,
} from 'vue-router'

import type { ServiceDefinition } from '@/services/utils'
import { token, createInjections } from '@/services/utils'
import type { Component } from 'vue'
import type { Router, RouteRecordRaw, NavigationGuard } from 'vue-router'

type Token = ReturnType<typeof token>

type VueApp = ReturnType<typeof createApp>
export type PluginDefinition = Parameters<VueApp['use']>
export type ComponentDefinition = [string, Component]

const $ = {
  app: token<(App: Component) => Promise<VueApp>>('vue.app'),
  router: token<Router>('vue.router'),

  components: token('vue.components'),
  plugins: token<PluginDefinition[]>('vue.plugins'),
  routes: token<RouteRecordRaw[]>('vue.routes'),
  routesLabel: token<RouteRecordRaw[]>('vue.routes.label'),
  navigationGuards: token<NavigationGuard[]>('vue.routes.navigation.guards'),
}
export const services = (app: Record<string, Token>): ServiceDefinition[] => {
  return [
    [$.app, {
      service: (
        components: ComponentDefinition[],
        plugins: PluginDefinition[],
      ) => {
        return async (App: Component) => {
          const app = createApp(App)

          plugins.forEach(([...args]) => {
            app.use(...args)
          })

          components.forEach(([name, component]: [string, Component]) => {
            app.component(name, component)
          })

          return app
        }
      },
      arguments: [
        $.components,
        $.plugins,
      ],
    }],
    [$.router, {
      service: (
        env: (str: 'KUMA_BASE_PATH') => string,
        routes: RouteRecordRaw[],
        guards: NavigationGuard[],
      ) => {
        const router = createRouter({
          history: createWebHistory(env('KUMA_BASE_PATH')),
          routes,
        })

        guards.forEach((item) => {
          if (typeof item === 'function') {
            router.beforeEach(item)
          }
        })

        return router
      },
      arguments: [
        app.env,
        $.routes,
        $.navigationGuards,
      ],
    }],

    [token('vue.plugins'), {
      service: (router) => {
        return [
          [router],
        ]
      },
      arguments: [
        $.router,
      ],
      labels: [
        $.plugins,
      ],
    }],

    // TODO(jc): Label decoration. Make it so we can decorate labels
    // Temporarily turn the routesLabel label into routes for decoration
    // purposes
    [$.routes, {
      service: (routes: RouteRecordRaw[]) => routes,
      arguments: [
        $.routesLabel,
      ],
    }],
    //

    // TODO(jc): Make it so we don't need to provide a [] empty default for
    // labels
    [token('application.routes.navigation.guards'), {
      service: () => {
        return []
      },
      labels: [
        $.navigationGuards,
      ],
    }],
  ]
}

export const TOKENS = $
export const [
  useRouter,
] = createInjections(
  TOKENS.router,
)
