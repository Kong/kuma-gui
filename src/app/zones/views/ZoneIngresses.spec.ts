import { flushPromises } from '@vue/test-utils'
import { render, screen } from '@testing-library/vue'
import userEvent from '@testing-library/user-event'

import ZoneIngresses from './ZoneIngresses.vue'
import { store, storeKey } from '@/store/store'
import { ClientConfigInterface } from '@/store/modules/config/config.types'
import * as config from '@/api/mock-data/config.json'

jest.mock('@/utilities/helpers', () => {
  const originalModule = jest.requireActual('@/utilities/helpers')

  return {
    __esModule: true,
    ...originalModule,
    humanReadableDate: (data: string) => data,
    rawReadableDate: (data: string) => data,
  }
})

function renderComponent(mode = 'standalone') {
  const clientConfig: ClientConfigInterface = { ...config, mode }
  store.state.config.clientConfig = clientConfig

  return render(ZoneIngresses, {
    global: {
      plugins: [[store, storeKey]],
    },
  })
}

describe('ZoneIngresses.vue', () => {
  it('renders snapshot when no multizone', async () => {
    const { container } = renderComponent()

    await flushPromises()

    expect(container).toMatchSnapshot()
  })

  it('renders snapshot when multizone', async () => {
    const { container } = renderComponent('global')

    await flushPromises()

    await screen.findByText(/ZoneIngressOverview/)

    expect(container).toMatchSnapshot()
  })

  it('renders zoneingress insights', async () => {
    renderComponent('global')

    await screen.findByText(/ZoneIngressOverview/)

    await userEvent.click(screen.getByText(/Zone Ingress Insights/))
    expect(screen.getByTestId('tab-container')).toMatchSnapshot()
  })
})
