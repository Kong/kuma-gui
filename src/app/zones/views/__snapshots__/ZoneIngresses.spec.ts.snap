// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoneIngresses.vue renders snapshot when multizone 1`] = `
<div>
  <div
    class="zoneingresses"
  >
    
    <!-- Zone CPs information for when Multicluster is enabled -->
    <div
      class="component-frame"
    >
      
      <div
        class="data-overview"
        data-testid="data-overview"
      >
        <div
          class="data-table-controls mb-2"
        >
          
          
          <button
            class="medium rounded primary k-button refresh-button"
            data-testid="refresh-button"
            data-v-6f773baa=""
            type="button"
          >
            
            <!---->
            
            
            <span
              class="refresh-icon custom-control-icon"
            >
              <svg
                viewBox="0 0 36 36"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#fff"
                  fill-rule="nonzero"
                >
                  <path
                    d="M18 5.5a12.465 12.465 0 00-8.118 2.995 1.5 1.5 0 001.847 2.363l.115-.095A9.437 9.437 0 0118 8.5l.272.004a9.487 9.487 0 019.07 7.75l.04.246H25a.5.5 0 00-.416.777l4 6a.5.5 0 00.832 0l4-6 .04-.072A.5.5 0 0033 16.5h-2.601l-.017-.15C29.567 10.2 24.294 5.5 18 5.5zM2.584 18.723l-.04.072A.5.5 0 003 19.5h2.6l.018.15C6.433 25.8 11.706 30.5 18 30.5c3.013 0 5.873-1.076 8.118-2.995a1.5 1.5 0 00-1.847-2.363l-.115.095A9.437 9.437 0 0118 27.5l-.272-.004a9.487 9.487 0 01-9.07-7.75l-.041-.246H11a.5.5 0 00.416-.777l-4-6a.5.5 0 00-.832 0l-4 6z"
                  />
                </g>
              </svg>
            </span>
            <span>
              Refresh
            </span>
            
            <!---->
          </button>
        </div>
        <div
          class="data-overview-content"
        >
          <!-- data -->
          <div
            class="data-overview-table"
          >
            <div
              class="k-table-container"
              data-testid="data-overview-table"
              data-v-4f741344=""
              is-clickable=""
            >
              <!---->
              <section
                class="k-table-wrapper"
                data-v-4f741344=""
              >
                <table
                  class="has-hover is-clickable k-table"
                  data-tableid="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                  data-v-4f741344=""
                >
                  <thead
                    class=""
                    data-v-4f741344=""
                  >
                    <tr
                      class=""
                      data-v-4f741344=""
                    >
                      
                      <th
                        class=""
                        data-v-4f741344=""
                      >
                        <span
                          class="d-flex align-items-center"
                          data-v-4f741344=""
                        >
                          
                          <span
                            class="sr-only"
                            data-v-4f741344=""
                          >
                            actions
                          </span>
                          
                          <!---->
                        </span>
                      </th>
                      <th
                        class=""
                        data-v-4f741344=""
                      >
                        <span
                          class="d-flex align-items-center"
                          data-v-4f741344=""
                        >
                          
                          <span
                            class=""
                            data-v-4f741344=""
                          >
                            Status
                          </span>
                          
                          <!---->
                        </span>
                      </th>
                      <th
                        class=""
                        data-v-4f741344=""
                      >
                        <span
                          class="d-flex align-items-center"
                          data-v-4f741344=""
                        >
                          
                          <span
                            class=""
                            data-v-4f741344=""
                          >
                            Name
                          </span>
                          
                          <!---->
                        </span>
                      </th>
                      
                    </tr>
                  </thead>
                  <tbody
                    data-v-4f741344=""
                  >
                    
                    <tr
                      class="is-selected"
                      data-v-4f741344=""
                      role="link"
                      tabindex="0"
                    >
                      
                      <td
                        class=""
                        data-v-4f741344=""
                      >
                        
                        
                        
                      </td>
                      <td
                        class=""
                        data-v-4f741344=""
                      >
                        
                        <div
                          class="entity-status is-online"
                        >
                          <span>
                            Online
                          </span>
                        </div>
                        
                      </td>
                      <td
                        class=""
                        data-v-4f741344=""
                      >
                        
                        
                        zone-ingress-1
                        
                        
                      </td>
                      
                    </tr>
                    
                  </tbody>
                </table>
                <!---->
              </section>
            </div>
            <div
              class="pagination"
            >
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div
        class="tab-container"
        data-testid="tab-container"
      >
        
        <header
          class="tab__header"
        >
          
          <div>
            <h3>
               Zone Ingress: zone-ingress-1
            </h3>
          </div>
          
        </header>
        <div
          class="tab__content-container"
        >
          <div
            class="k-tabs"
            data-v-b0b5f6ae=""
          >
            <ul
              aria-label="Tabs"
              data-v-b0b5f6ae=""
              role="tablist"
            >
              
              <li
                aria-controls="panel-0"
                aria-selected="true"
                class="active tab-item"
                data-v-b0b5f6ae=""
                id="overview-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-b0b5f6ae=""
                >
                  
                  Overview
                  
                </a>
              </li>
              <li
                aria-controls="panel-1"
                aria-selected="false"
                class="tab-item"
                data-v-b0b5f6ae=""
                id="insights-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-b0b5f6ae=""
                >
                  
                  Zone Ingress Insights
                  
                </a>
              </li>
              <li
                aria-controls="panel-2"
                aria-selected="false"
                class="tab-item"
                data-v-b0b5f6ae=""
                id="xds-configuration-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-b0b5f6ae=""
                >
                  
                  XDS Configuration
                  
                </a>
              </li>
              <li
                aria-controls="panel-3"
                aria-selected="false"
                class="tab-item"
                data-v-b0b5f6ae=""
                id="envoy-stats-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-b0b5f6ae=""
                >
                  
                  Stats
                  
                </a>
              </li>
              <li
                aria-controls="panel-4"
                aria-selected="false"
                class="tab-item"
                data-v-b0b5f6ae=""
                id="envoy-clusters-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-b0b5f6ae=""
                >
                  
                  Clusters
                  
                </a>
              </li>
              
            </ul>
            
            <div
              aria-labelledby="overview-tab"
              class="tab-container"
              data-v-b0b5f6ae=""
              id="panel-0"
              role="tabpanel"
              tabindex="0"
            >
              
              
              <div>
                <div
                  class="label-list-content"
                >
                  <section
                    aria-describedby="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                    class="noBorder kong-card"
                    data-v-1f0d90ac=""
                  >
                    <!---->
                    <!---->
                    <div
                      class="k-card-content d-flex"
                      data-v-1f0d90ac=""
                    >
                      <div
                        class="k-card-body"
                        data-v-1f0d90ac=""
                        id="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                      >
                        
                        <div
                          class="label-list__col-wrapper multi-col"
                        >
                          
                          <div>
                            <ul>
                              
                              <li>
                                <h4>
                                  type
                                </h4>
                                <p>
                                  ZoneIngressOverview
                                </p>
                              </li>
                              <li>
                                <h4>
                                  name
                                </h4>
                                <p>
                                  zone-ingress-1
                                </p>
                              </li>
                              
                            </ul>
                          </div>
                          
                        </div>
                        
                      </div>
                      <!---->
                    </div>
                  </section>
                </div>
              </div>
              
              
            </div>
            <div
              aria-labelledby="insights-tab"
              class="tab-container"
              data-v-b0b5f6ae=""
              id="panel-1"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="xds-configuration-tab"
              class="tab-container"
              data-v-b0b5f6ae=""
              id="panel-2"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="envoy-stats-tab"
              class="tab-container"
              data-v-b0b5f6ae=""
              id="panel-3"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="envoy-clusters-tab"
              class="tab-container"
              data-v-b0b5f6ae=""
              id="panel-4"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            
          </div>
        </div>
        
      </div>
      
    </div>
    
  </div>
</div>
`;

exports[`ZoneIngresses.vue renders snapshot when no multizone 1`] = `
<div>
  <div
    class="zoneingresses"
  >
    <section
      class="empty-state-wrapper"
      data-v-5b55eb64=""
    >
      <div
        class="empty-state-title"
        data-v-5b55eb64=""
      >
        <!---->
        <div
          class="k-empty-state-title-header mt-4 mb-4"
          data-v-5b55eb64=""
        >
          
          <span
            class="kong-icon--centered kong-icon-dangerCircleOutline kong-icon"
            data-v-71cfcfc4=""
          >
            <svg
              fill="none"
              height="42"
              role="img"
              viewBox="0 0 17 17"
              width="42"
              xmlns="http://www.w3.org/2000/svg"
            >
              
  
              <title>
                Danger
              </title>
              
  
              <path
                clip-rule="evenodd"
                d="M16.5 8a8 8 0 11-16 0 8 8 0 0116 0zm-2 0a6 6 0 11-12 0 6 6 0 0112 0zm-7-3.5A.5.5 0 018 4h1a.5.5 0 01.5.5v4A.5.5 0 019 9H8a.5.5 0 01-.5-.5v-4zM8 10a.5.5 0 00-.5.5v1a.5.5 0 00.5.5h1a.5.5 0 00.5-.5v-1A.5.5 0 009 10H8z"
                fill="#A3BBCC"
                fill-rule="evenodd"
              />
              

            </svg>
            

          </span>
          <p>
            Kuma is running in Standalone mode.
          </p>
          
        </div>
      </div>
      <div
        class="empty-state-content"
        data-v-5b55eb64=""
      >
        <div
          class="k-empty-state-message mb-6"
          data-v-5b55eb64=""
        >
          
          <p>
             To access this page, you must be running in 
            <strong>
              Multi-Zone
            </strong>
             mode. 
          </p>
          
        </div>
        <div
          class="k-empty-state-cta"
          data-v-5b55eb64=""
        >
          
          <a
            class="medium rounded primary k-button"
            data-v-6f773baa=""
            href="https://kuma.io/docs/latest/documentation/deployments/"
            target="_blank"
            type="button"
          >
            
            <!---->
            
            
             Learn More 
            
            <!---->
          </a>
          
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`ZoneIngresses.vue renders zoneingress insights 1`] = `
<div
  class="tab-container"
  data-testid="tab-container"
>
  
  <header
    class="tab__header"
  >
    
    <div>
      <h3>
         Zone Ingress: zone-ingress-1
      </h3>
    </div>
    
  </header>
  <div
    class="tab__content-container"
  >
    <div
      class="k-tabs"
      data-v-b0b5f6ae=""
    >
      <ul
        aria-label="Tabs"
        data-v-b0b5f6ae=""
        role="tablist"
      >
        
        <li
          aria-controls="panel-0"
          aria-selected="false"
          class="tab-item"
          data-v-b0b5f6ae=""
          id="overview-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-b0b5f6ae=""
          >
            
            Overview
            
          </a>
        </li>
        <li
          aria-controls="panel-1"
          aria-selected="true"
          class="active tab-item"
          data-v-b0b5f6ae=""
          id="insights-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-b0b5f6ae=""
          >
            
            Zone Ingress Insights
            
          </a>
        </li>
        <li
          aria-controls="panel-2"
          aria-selected="false"
          class="tab-item"
          data-v-b0b5f6ae=""
          id="xds-configuration-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-b0b5f6ae=""
          >
            
            XDS Configuration
            
          </a>
        </li>
        <li
          aria-controls="panel-3"
          aria-selected="false"
          class="tab-item"
          data-v-b0b5f6ae=""
          id="envoy-stats-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-b0b5f6ae=""
          >
            
            Stats
            
          </a>
        </li>
        <li
          aria-controls="panel-4"
          aria-selected="false"
          class="tab-item"
          data-v-b0b5f6ae=""
          id="envoy-clusters-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
            data-v-b0b5f6ae=""
          >
            
            Clusters
            
          </a>
        </li>
        
      </ul>
      
      <div
        aria-labelledby="overview-tab"
        class="tab-container"
        data-v-b0b5f6ae=""
        id="panel-0"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="insights-tab"
        class="tab-container"
        data-v-b0b5f6ae=""
        id="panel-1"
        role="tabpanel"
        tabindex="0"
      >
        
        
        <section
          aria-describedby="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
          class="noBorder kong-card"
          data-v-1f0d90ac=""
        >
          <!---->
          <!---->
          <div
            class="k-card-content d-flex"
            data-v-1f0d90ac=""
          >
            <div
              class="k-card-body"
              data-v-1f0d90ac=""
              id="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
            >
              
              <ul
                class="accordion"
              >
                
                
                <li
                  class="accordion-item relative border-b active"
                >
                  <button
                    aria-expanded="true"
                    class="accordion-item-header"
                    type="button"
                  >
                    
                    <h4
                      class="text-lg font-medium"
                    >
                      <span
                        class="color-green-400"
                      >
                         Connect time: 2021-07-13T08:41:04.556796688Z
                      </span>
                      <!--v-if-->
                    </h4>
                    
                  </button>
                  <transition-stub
                    appear="false"
                    css="true"
                    name="accordion"
                    persisted="false"
                  >
                    <div
                      class="px-4 py-1"
                    >
                      
                      <div>
                        <div>
                          <h5
                            class="overview-tertiary-title"
                          >
                             General Information: 
                          </h5>
                          <ul>
                            <!--v-if-->
                            <li>
                              <strong>
                                Control Plane Instance ID:
                              </strong>
                                
                              <span
                                class="mono"
                              >
                                kuma-control-plane-84f5589874-nmspq-0867
                              </span>
                            </li>
                            <li>
                              <strong>
                                Last Connected:
                              </strong>
                                2021-07-13T08:41:04.556796688Z
                            </li>
                            <!--v-if-->
                          </ul>
                        </div>
                        <div>
                          <ul
                            class="overview-stat-grid"
                          >
                            
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                Cds: 
                              </h6>
                              <ul>
                                
                                <li>
                                  <strong>
                                    Responses Sent:
                                  </strong>
                                    
                                  <span
                                    class="mono"
                                  >
                                    3
                                  </span>
                                </li>
                                <li>
                                  <strong>
                                    Responses Acknowledged:
                                  </strong>
                                    
                                  <span
                                    class="mono"
                                  >
                                    3
                                  </span>
                                </li>
                                
                              </ul>
                            </li>
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                Eds: 
                              </h6>
                              <ul>
                                
                                <li>
                                  <strong>
                                    Responses Sent:
                                  </strong>
                                    
                                  <span
                                    class="mono"
                                  >
                                    2
                                  </span>
                                </li>
                                <li>
                                  <strong>
                                    Responses Acknowledged:
                                  </strong>
                                    
                                  <span
                                    class="mono"
                                  >
                                    3
                                  </span>
                                </li>
                                
                              </ul>
                            </li>
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                Lds: 
                              </h6>
                              <ul>
                                
                                <li>
                                  <strong>
                                    Responses Sent:
                                  </strong>
                                    
                                  <span
                                    class="mono"
                                  >
                                    3
                                  </span>
                                </li>
                                <li>
                                  <strong>
                                    Responses Acknowledged:
                                  </strong>
                                    
                                  <span
                                    class="mono"
                                  >
                                    3
                                  </span>
                                </li>
                                
                              </ul>
                            </li>
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                Rds: 
                              </h6>
                              <ul>
                                
                                
                              </ul>
                            </li>
                            
                          </ul>
                        </div>
                      </div>
                      
                    </div>
                  </transition-stub>
                </li>
                
                
              </ul>
              
            </div>
            <!---->
          </div>
        </section>
        
        
      </div>
      <div
        aria-labelledby="xds-configuration-tab"
        class="tab-container"
        data-v-b0b5f6ae=""
        id="panel-2"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="envoy-stats-tab"
        class="tab-container"
        data-v-b0b5f6ae=""
        id="panel-3"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="envoy-clusters-tab"
        class="tab-container"
        data-v-b0b5f6ae=""
        id="panel-4"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      
    </div>
  </div>
  
</div>
`;
